desc/compute/utils.py

```diff
@@ -538,6 +541,14 @@ def surface_integrals(grid, q=jnp.array([1]), surface_label="rho"):
     q = jnp.atleast_1d(q)
     nodes, unique_idx, ds, max_surface_val = _get_grid_surface(grid, surface_label)
 
+    if max_surface:
+        max_rho = grid.nodes[grid.unique_rho_idx[-1], 0]
+        idx = np.nonzero(grid.nodes[:, 0] == max_rho)[0]
+        q = q[idx]
+        nodes = nodes[idx]
+        unique_idx = (unique_idx * nodes.size / grid.num_nodes).astype(int)
```

Comment:
can you explain this line?

Reply:
After extracting only the nodes at the max rho surface in line 548, `unique_idx` no longer corresponds to the correct values in the `nodes` array. `nodes` now only corresponds to nodes on a single flux surface, but `unique_idx` is still pointing to indices in the full list from `grid.nodes`. Based on how the grids are sorted this is corrected by dividing all of the indices by the number of rho surfaces. 

Multiplying by `nodes.size / grid.num_nodes` is equivalent to dividing by `grid.num_rho`, the number of rho surfaces. I made that change to make it more clear.